test/lib/ansible_test/_data/requirements/constraints.txt

```diff
@@ -58,7 +58,7 @@ isort == 4.3.15
 lazy-object-proxy == 1.3.1
 mccabe == 0.6.1
 pylint == 2.3.1
-typed-ast == 1.4.0  # 1.4.0 is required to compile on Python 3.8
+typed-ast == 1.4.1  # 1.4.1 is required to compile on Python 3.9
```

Comment:
I'd make the changes for pylint in a separate PR, since they're likely going to be backported differently than the other changes in this PR.

Reply:
I can do it in a separate commit in the same PR, but doing a separate PR will be hard since they're needed for tests to pass ;)

Reply:
I'd like to see the pylint changes in a separate PR that we merge first, then re-run tests on this PR before merging.